### PR TITLE
Tweak search bar styles in homepage

### LINKED ISSUE #728: Search bar styles in homepage
## Description
On desktop, the search bar does not have the height and the CTA style defined in mockups. On mobile, search input has two colors in the focus state, and the screen zooms in, focusing on the input element.

CleanShot 2022-01-27 at 10 58 06@2x

https://user-images.githubusercontent.com/895819/151374757-b60be9a6-2ab1-4103-a327-6f866434e820.mov

## Additional context
Previously in the mockups, the search bar had a dark border to meet the A11Y requirements, but that was a misunderstanding of the criterion. The element now has no border in its resting state.

### Figma references
- [**Home page on Desktop** ](https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Openverse-Releases?node-id=1423%3A20197)
- [**Home page on Mobile**](https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Openverse-Releases?node-id=1424%3A21449)

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Partially fixes #728 by @panchovm 

## Description

This was a bit complicated but it should do the following on the homepage without altering the searchbar style on other pages:
- Adds the button with "Search" text on desktop
- Makes the search icon button pink with white text on mobile
- Open Creative Commons link on disclaimer text in a new tab as it doesn't work in the iframe 

## Testing Instructions

Run the project and see the homepage search bar. Click on the link of the CC licenses and see it opens in a new tab.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
> Adds the button with "Search" text on desktop

I'm not remembering where this idea was discussed. @panchovm do you have context on this?

**Comment by zackkrida** [comment]
This looks _really nice_ 😄 I made a small related PR (#819) to fix mobile safari, I _think_ these 2 PRs combined fix the entire issue.

**Comment by panchovm** [comment]
We did not discuss the search button specifically but the layout and image gallery on the proposal discussion (#592). Do you have some concerns about replacing the search icon with the `Search` label?

File: test/unit/specs/components/VHeader/SearchBar/search-bar.spec.js [hunk 1]
```diff
@@ -1,29 +1,48 @@
...
+      expect(inputElement.tagName).toBe('INPUT')
+      expect(inputElement).toHaveAttribute('type', 'search')
+      expect(inputElement).toHaveAttribute('placeholder', 'Enter search query')
+    }
+  )
+
+  it.each(sizes)(
+    'renders a button with type="submit", ARIA label and SR text (%s size)',
+    (size) => {
+      useMatchHomeRoute.mockImplementation(() => false)
```
Comment (krysal) [inline test/unit/specs/components/VHeader/SearchBar/search-bar.spec.js:37]: I'm not sure is this value is correct but the following doesn't seem to work.
```js
useMatchHomeRoute.mockImplementation(() => {matches: false})
```

Comment (sarayourfriend) [reply test/unit/specs/components/VHeader/SearchBar/search-bar.spec.js:37]: `matches` needs to be a `Ref` object to match the return value of the actual function.

File: src/components/VHeader/VSearchBar/VSearchButton.vue [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
+  <div>
+    <VButton
+      v-if="isHomeRoute && isMinScreenMd"
+      v-bind="$attrs"
+      :aria-label="$t('search.search')"
+      class="inline-block cta-search-button font-semibold h-full text-2xl"
+      v-on="$listeners"
+      >{{ $t('search.search') }}</VButton
+    >
+    <VIconButton
```
Comment (sarayourfriend) [inline src/components/VHeader/VSearchBar/VSearchButton.vue:11]: It _kind of_ seems like these should be combinable in some way... Like `VIconButton` should just be an implementation of `VButton` or something?

Probably for a follow up issue 👍

**Review by sarayourfriend — APPROVED** [review]
Looks good to me! I noticed one minor issue with RTL where the search button extends far enough now to be covered by the featured images. Not sure what is causing it but it's very minor. If it takes more than 2 minutes to figure it out I would just open another issue for it to fix later. It doesn't break the functionality of the page and is only at a very specific narrow range of viewport widths.

Captura de Tela 2022-02-09 às 11 18 59

**Review by obulat — APPROVED** [review]

**Comment by panchovm** [comment]
To @sarayourfriend's point, I think we can push the gallery slightly towards its left. Here is a screenshot of a `1370px width` view where the left area covers the gallery. It seems reasonable to follow the same solution for this breakpoint.

CleanShot 2022-02-09 at 14 52 15@2x

File: test/unit/specs/components/VHeader/SearchBar/search-bar.spec.js (hunk 1 above)
Comment (krysal) [reply test/unit/specs/components/VHeader/SearchBar/search-bar.spec.js:37]: Yes, I tried with a `ref` too but strangely it wasn't passing either 🤔 I hope to find the reason later.

**Comment by krysal** [comment]
It took more than two minutes so I'd go with the @sarayourfriend suggested. Weirdly the overlay only happens in RTL mode.
I also added a 1px border bottom to the button because it was showing a white space in certain screen sizes.

**Pull request merged**